JavaScript/chapter02/p03_delete_middle_node/avc278.js

```diff
@@ -0,0 +1,66 @@
+// Delete Middle Node: Implement an algorithm to delete a node in the middle (i.e. any node but the first and last node,
+// not necessarily the exact middle) of a singly linked list, given only access to that node.
+// EXAMPLE:
+// Input: the node c from the linked list a => b => c => d => e => f
+// Result: nothing is returned, but the new linked list looks like a => b => d => e => f
+
+const assert = require("assert");
+
+class LinkedListNode {
+  constructor(val, next) {
+    this.val = val === undefined ? null : val;
+    this.next = next === undefined ? null : next;
+  }
+}
+
+const arrayToLinkedList = (arr) => {
+  let tail = null;
+  for (let i = arr.length - 1; i >= 0; i--) {
+    tail = new LinkedListNode(arr[i], tail);
+  }
+  return tail;
+};
+
+const compareLinkedLists = (A, B) => {
+  if (!A && !B) return true;
+
+  while (A !== null && B !== null) {
+    if (A.val !== B.val) return false;
+    A = A.next;
+    B = B.next;
+  }
+  return !((A && !B) || (!A && B));
```

Comment:
```suggestion
  return A !== null && B !== nulll;
```

Reply:
Reading back on the explanation in #65, I do think we want to return true when we have exhausted A and B.
So `return A === null && B === null` would be valid here, or just `return !A && !B`.

Reply:
err... yes, i messed up the negations ^^;;

Reply:
Oh btw, while we're here, would you mind if I made a separate directory in chapter 2 for re-usable functions/classes just to DRY things out?
Something like `JavaScript/chapter02/utils/avc278.js`

Reply:
creating a directory to host re-usable functions/classes sounds good!

in some circles, the name "utils" is frowned upon (e.g. https://softwareengineering.stackexchange.com/a/175190, https://www.vojtechruzicka.com/avoid-utility-classes/), so `JavaScript/lib/avc278/linkedlist.js` may be a better location.

Reply:
Oo sounds good to me. Keep an eye out for it in my next PR.
I'll also sneak in refactors to the previous solutions in chapter 2.